dev/provider_packages/get_provider_info_TEMPLATE.py.jinja2

```diff
@@ -0,0 +1,13 @@
+import yaml
+
+try:
+    import importlib.resources as importlib_resources
+except ImportError:
+    # Try backported to PY<37 `importlib_resources`.
+    import importlib_resources
+
+
+def get_provider_info():
+    return yaml.safe_load("""
+{{ PROVIDER_INFO }}
+""")
```

Comment:
Do need to YAML load this - could we not return a python dict directly?

Reply:
Actually: bigger question: Do we ship and install the provider.yaml with the dist, or do we just use this function.

Not shipping it (if all the info is contained in here anyway) might be a nice/easy way to distinguish source vs installed provider.

Reply:
I agree. We can serialize this object to Python. Here is a helpful function, it is tested and works with all primitive types. 
https://gist.github.com/mik-laj/3ae253d5394b34447dcbd426fd19b389#file-convert-py-L31

Reply:
> Actually: bigger question: Do we ship and install the provider.yaml with the dist, or do we just use this function.
> 
> Not shipping it (if all the info is contained in here anyway) might be a nice/easy way to distinguish source vs installed provider.

Not shipping - absolutely. I did not intend it (it was there in the first version but when I switched to entry_points precisely for this - being able to distinguish "package" vs. "local source".

Reply:
That's possible more complex than we need @mik-laj  -- could we just use `{{ PROVIDER_INFO }}` if we did a `yaml.load()` in the template scripts -- i.e. `PROVIDER_INFO` would be a python dict, and the default repr is valid python code.

Reply:
> I agree. We can serialize this object to Python. Here is a helpful function, it is tested and works with all primitive types.
> https://gist.github.com/mik-laj/3ae253d5394b34447dcbd426fd19b389#file-convert-py-L31

That was my worry about generating python dict out of the metadata (we need to generate valid python code). I am not sure if it is worth the hassle.

This code is generated on the flight, nobody will really see it, it's a very small speed improvement (if at all). 

 We are basically trading parsing of python code with parsing of yaml code.

WDYT @mik-laj @ashb -> do you really see a benefit of doing it (knowing that we need to add new code to handle it ?)

Reply:
Also with YAML we could have comments and other stuff that will be retained in such generated class (in case someone actually does look there). Not that we need it now, but I can imagine it can be useful to keep it).

Reply:
We don't have to worry about the validitiy of the generated code at run-time - tests should catch any errors. But I can't see how we'd get any: safe_load only supports "primitive"/native types that python can express anyway.

That YAML file is for computers -- any comments that a user might need to see should go in to docstrings for that provider.

It just feels odd to YAML parse a static string given this file is generated anyway.

Reply:
Oh and with .pyc files we pay that parsing time only once, YAML parse we'd pay every time.

Reply:
> Oh and with .pyc files we pay that parsing time only once, YAML parse we'd pay every time.

True. Let me try.